### PR TITLE
DEX-270 Tests for dynamic order-fee mode

### PR DESCRIPTION
**Pull request by frekensnork**
-- added tests for settings and REST handlers

### CODE REVIEW

File: dex-it/src/test/scala/com/wavesplatform/it/sync/OrderFeeTestSuite.scala [hunk 1]
```diff
@@ -0,0 +1,430 @@
...
+          price = 50000L,
+          fee = 150L,
+          version = 3: Byte,
+          matcherFeeAssetId = IssuedAsset(BtcId)
+        )
+      )
+      node.reservedBalance(bob)(BtcId.toString) shouldEqual 50150L
+      node.reservedBalance(bob).keys shouldNot contain("WAVES")
+      node.cancelAllOrders(bob)
+      node.deleteRate(IssuedAsset(BtcId))
```
Comment (DrBlast) [inline dex-it/src/test/scala/com/wavesplatform/it/sync/OrderFeeTestSuite.scala]: add: bob has some amount for fee, but not enough to pay fee, additional fee should be withdraw after order fill

File: dex-it/src/test/scala/com/wavesplatform/it/sync/OrderFeeTestSuite.scala [hunk 2]
```diff
@@ -0,0 +1,430 @@
...
+          pair = assetPair,
+          orderType = OrderType.SELL,
+          amount = 1.waves,
+          price = 50000L,
+          fee = 1920L,
+          version = 3,
+          matcherFeeAssetId = IssuedAsset(EthId)
+        )
+      ).message.id
+      node.waitOrderStatus(assetPair, bobOrderId, "Filled", 500.millis)
```
Comment (DrBlast) [inline dex-it/src/test/scala/com/wavesplatform/it/sync/OrderFeeTestSuite.scala]: check balances, correct amount of eth should be withdrawn to pay fee

File: dex-it/src/test/scala/com/wavesplatform/it/sync/OrderFeeTestSuite.scala [hunk 3]
```diff
@@ -0,0 +1,430 @@
...
+          amount = 1.waves,
+          price = 50000L,
+          fee = 150L,
+          version = 3: Byte,
+          matcherFeeAssetId = IssuedAsset(BtcId)
+        )
+      ).message.id
+      node.waitOrderStatus(assetPair, aliceOrderId, "Filled", 500.millis)
+      node.waitOrderInBlockchain(bobSecondOrderId)
+      node.deleteRate(IssuedAsset(BtcId))
```
Comment (DrBlast) [inline dex-it/src/test/scala/com/wavesplatform/it/sync/OrderFeeTestSuite.scala]: check balances

File: dex-it/src/test/scala/com/wavesplatform/it/sync/OrderFeeTestSuite.scala [hunk 4]
```diff
@@ -0,0 +1,430 @@
...
+          pair = assetPair,
+          orderType = OrderType.SELL,
+          amount = 1.waves,
+          price = 50000L,
+          fee = 1920L,
+          version = 3,
+          matcherFeeAssetId = IssuedAsset(EthId)
+        )
+      ).message.id
+      Array(bobOrderId, aliceOrderId)
```
Comment (DrBlast) [inline dex-it/src/test/scala/com/wavesplatform/it/sync/OrderFeeTestSuite.scala]: remove default waittime

File: dex-it/src/test/scala/com/wavesplatform/it/sync/OrderFeeTestSuite.scala [hunk 5]
```diff
@@ -0,0 +1,430 @@
...
+        .foreach(orderId => node.waitOrderInBlockchain(orderId))
+      node.cancelOrder(alice, assetPair, aliceOrderId).status shouldBe "OrderCanceled"
+      node.reservedBalance(alice).keys.size shouldBe 0
+      node.assertAssetBalance(alice.address, EthId.toString, aliceEthBalance - 960L)
+      Array(BtcId, EthId)
+        .foreach(assetId => node.deleteRate(IssuedAsset(assetId)))
+    }
+  }
```
Comment (DrBlast) [inline dex-it/src/test/scala/com/wavesplatform/it/sync/OrderFeeTestSuite.scala]: change rate when order placed. check balances

File: dex-it/src/test/scala/com/wavesplatform/it/sync/smartcontracts/ExtraFeeTestSuite.scala [hunk 6]
```diff
@@ -117,6 +149,41 @@ class ExtraFeeTestSuite extends MatcherSuiteBase {
         }
       }
     }
+
+    "with non-waves asset fee with one Smart Account and one Smart Asset" in {
```
[the remark this answers is not in the document]
Comment (frekensnork) [reply dex-it/src/test/scala/com/wavesplatform/it/sync/smartcontracts/ExtraFeeTestSuite.scala]: there is a test in 201th string

**Pull request merged**